pkg/agent/route/route_windows.go

```diff
@@ -88,6 +97,10 @@ func (c *Client) Reconcile(podCIDRs []string) error {
 			c.hostRoutes.Store(dst, rt)
 			continue
 		}
+		// If the route is not for uplink interface, ignore it.
+		if c.nodeConfig.UplinkNetConfig != nil && rt.LinkIndex != c.nodeConfig.UplinkNetConfig.Index {
```

Comment:
I just found this condition is not correct. `listRoutes` only returns routes on antrea-gw0 or br-int, so this is going to be always true. And I don't see why we need to check if it's for uplink interface here. We have migrated  all routes from uplink to br-int for noEncap case.

I also wonder how we know which route can be deleted as there could be user configured routes on br-int for noEncap case. I checked linux case, it cleans orphaned routes on antrea-gw0 only, which seems working for encap only but at least won't cause problem. I would suggest keep windows the same, i.e. don't try to clean up routes on br-int in `Reconcile` until there is a way to distinguish user configured routes.

Reply:
Removed.